cache_archive.go

```diff
@@ -112,19 +112,24 @@ func (a *Archive) WriteHeader(descriptor map[string]string, descriptorPth string
 		return err
 	}
 
-	closingHeader := &tar.Header{
+	return a.writeData(b, descriptorPth)
+}
+
+// writeData writes the byte array into the archive.
+func (a *Archive) writeData(data []byte, descriptorPth string) error {
```

Comment:
writeData is misleading, as it contains predefined tar header, so with this method it is only possible to write regular file into the archive. maybe it makes no other sense but it is great to know that there are many other types than can be included in an archive:
https://golang.org/pkg/archive/tar/#pkg-constants

Reply:
so handle this comment as a note